matrix/validate_sudoku_board.py

```diff
@@ -0,0 +1,101 @@
+"""
+LeetCode 36. Valid Sudoku
+https://leetcode.com/problems/valid-sudoku/
+https://en.wikipedia.org/wiki/Sudoku
+
+Determine if a 9 x 9 Sudoku board is valid. Only the filled cells need to be
+validated according to the following rules:
+
+- Each row must contain the digits 1-9 without repetition.
+- Each column must contain the digits 1-9 without repetition.
+- Each of the nine 3 x 3 sub-boxes of the grid must contain the digits 1-9
+  without repetition.
+
+Note:
+
+A Sudoku board (partially filled) could be valid but is not necessarily
+solvable.
+
+Only the filled cells need to be validated according to the mentioned rules.
+"""
+
+from collections import defaultdict
+
+NUM_ROWS = 9
+NUM_COLS = 9
+EMPTY_CELL = "."
+IS_VALID = True
+
+
+def is_valid_sudoku_board(sudoku_board: list[list[str]]) -> bool:
```

Comment:
Do this with int, not str and usr `0` for empty then in your printing code replace `0` with ` `.
```suggestion
def is_valid_sudoku_board(sudoku_board: list[list[int]]) -> bool:
```